dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/pdfform/DefaultPdfDataEntryFormService.java

```diff
@@ -813,13 +806,14 @@ private Period setPeriodDateRange( PeriodType periodType )
         int startYear = currYear - PERIODRANGE_PREVYEARS;
         int endYear = currYear + PERIODRANGE_FUTUREYEARS;
 
-        if ( periodType.getName().equals( QuarterlyPeriodType.NAME )
-            || periodType.getName().equals( SixMonthlyPeriodType.NAME )
-            || periodType.getName().equals( SixMonthlyAprilPeriodType.NAME )
-            || periodType.getName().equals( YearlyPeriodType.NAME )
-            || periodType.getName().equals( FinancialAprilPeriodType.NAME )
-            || periodType.getName().equals( FinancialJulyPeriodType.NAME )
-            || periodType.getName().equals( FinancialOctoberPeriodType.NAME ) )
+        if ( periodType.getName().equals( PeriodTypeEnum.QUARTERLY.getName() )
+            || periodType.getName().equals( PeriodTypeEnum.SIX_MONTHLY.getName() )
+            || periodType.getName().equals( PeriodTypeEnum.SIX_MONTHLY_APRIL.getName() )
+            || periodType.getName().equals( PeriodTypeEnum.YEARLY.getName() )
+            || periodType.getName().equals( PeriodTypeEnum.FINANCIAL_APRIL.getName() )
+            || periodType.getName().equals( PeriodTypeEnum.FINANCIAL_JULY.getName() )
+            || periodType.getName().equals( PeriodTypeEnum.FINANCIAL_OCT.getName() )
+            || periodType.getName().equals( PeriodTypeEnum.FINANCIAL_NOV.getName() ) )
```

Comment:
First impression here is that we should compare enum values not strings.

Reply:
Yes agreed. That entire method and class is very strange and needs a complete refactor and was some external contribution at the time, so I just didn't want to start fiddling with it.